auth-api/src/auth_api/services/api_gateway.py

```diff
@@ -0,0 +1,126 @@
+        consumer_endpoint: str = current_app.config.get('API_GW_CONSUMERS_API_URL')
+        gw_api_key = current_app.config.get('API_GW_KEY') if env == 'prod' else current_app.config.get(
+            'API_GW_NON_PROD_KEY')
+        email = cls._get_email_id(org_id)
+
+        if not org.has_api_access:  # If the account doesn't have api access, add it
+            client_rep = generate_client_representation(org_id, current_app.config.get('API_GW_KC_CLIENT_ID_PATTERN'))
+            KeycloakService.create_client(client_rep)
+            service_account = KeycloakService.get_service_account_user(client_rep.get('id'))
+            KeycloakService.add_user_to_group(service_account.get('id'), GROUP_API_GW_USERS)
+            KeycloakService.add_user_to_group(service_account.get('id'), GROUP_ACCOUNT_HOLDERS)
+
+            # Create a consumer with the keycloak client id and secret
+            create_consumer_payload = dict(email=email,
+                                           firstName=org.name,
+                                           lastName=org.branch_name or 'BCR',
+                                           userName=org.name,
+                                           clientId=client_rep.get('clientId'),
+                                           clientSecret=client_rep.get('secret'),
+                                           apiAccess=['ALL_API'],
+                                           apiKeyName=name)
+            api_key_response = RestService.post(
+                f'{consumer_endpoint}/mc/v1/consumers',
+                additional_headers={'x-apikey': gw_api_key},
+                data=create_consumer_payload,
+                generate_token=False
+            )
+            org.has_api_access = True
+            org.save()
+        else:
+            # Create additional API Key if a consumer exists
+            api_key_response = RestService.post(
+                f'{consumer_endpoint}/mc/v1/consumers/{email}/apikeys',
+                additional_headers={'x-apikey': gw_api_key},
+                data=dict(
+                    apiAccess=['ALL_API'],
+                    apiKeyName=name
+                ),
+                generate_token=False
+            )
+
+        return api_key_response.json()
+
+    @classmethod
+    def revoke_key(cls, org_id: int, api_key: str):
+        """Revoke api key."""
+        current_app.logger.debug('<revoke_key ')
+        consumer_endpoint: str = current_app.config.get('API_GW_CONSUMERS_API_URL')
+        gw_api_key: str = current_app.config.get('API_GW_KEY')
+        email_id = cls._get_email_id(org_id)
+
+        RestService.patch(
+            f'{consumer_endpoint}/mc/v1/consumers/{email_id}/apiKeys/{api_key}?action=removeAccess',
+            additional_headers={'x-apikey': gw_api_key},
+            data=dict(apiAccess='ALL_API'),
+            generate_token=False
+        )
+
+    @classmethod
+    def get_api_keys(cls, org_id: int) -> List[Dict[str, any]]:
```

Comment:
no auth check needed here? i think its again system..

Reply:
Yea, didnt add any auth check as this will be a manual step for a while.